icu4c/source/common/ustrcase.cpp

```diff
@@ -477,32 +479,65 @@ ustrcase_internalToTitle(int32_t caseLocale, uint32_t options, BreakIterator *it
                     return 0;
                 }
 
+                char buffer[30000];
+                int32_t bufferLength;
+
+
+                std::cout << "<<<<<<<<<<<<<<<<<<<<<<" << std::endl;
+                std::cout << "prev = " << prev << std::endl;
+                std::cout << "index = " << index << std::endl;
+                std::cout << "destIndex = " << destIndex << std::endl;
+                std::cout << "titleStart = " << titleStart << std::endl;
+                std::cout << "titleLimit = " << titleLimit << std::endl;
+                
+                UnicodeString dbg(dest, destIndex, destCapacity);
+                bufferLength = dbg.extract(0, dbg.length(), buffer, sizeof(buffer));
+                std::cout << "dest (dbg) = " << buffer << std::endl;
+
+                std::cout << "csc.cpStart = " << csc.cpStart << std::endl;
+                std::cout << "csc.cpLimit = " << csc.cpLimit << std::endl;
+                std::cout << "c = " << c << std::endl;
+
                 /* Special case Dutch IJ titlecasing */
                 if (titleStart+1 < index &&
```

Comment:
I needed to use `titleLimit` in the UTF-8 because when the first character is U+00CD (pre-composed capital I with acute), `titleStart + 1 != titleLimit`.